src/modules/control_allocator/ControlAllocation/ControlAllocationSequentialDesaturationTest.cpp

```diff
@@ -0,0 +1,383 @@
+	// Negative, because +z is "downward".
+	constexpr float THRUST_Z_TOTAL{-0.75};
+	// This is low enough to not saturate the motors.
+	constexpr float PITCH_CONTROL_SP{0.1};
+	control_sp(ControlAllocation::ControlAxis::ROLL) = 0;
+	control_sp(ControlAllocation::ControlAxis::PITCH) = PITCH_CONTROL_SP;
+	control_sp(ControlAllocation::ControlAxis::YAW) = 0;
+	control_sp(ControlAllocation::ControlAxis::THRUST_X) = 0;
+	control_sp(ControlAllocation::ControlAxis::THRUST_Y) = 0;
+	control_sp(ControlAllocation::ControlAxis::THRUST_Z) = THRUST_Z_TOTAL;
+	allocator.setControlSetpoint(control_sp);
+
+	// Since MC_AIRMODE was not set explicitly, assume airmode is disabled.
+	allocator.allocate();
+
+	const auto &actuator_sp = allocator.getActuatorSetpoint();
+	// This value is based off of the effectiveness matrix. If the effectiveness matrix is changed,
+	// this will need to be changed.
+	constexpr float PITCH_DIFF_PER_MOTOR{PITCH_CONTROL_SP / 4};
+	// At control set point, there will be 2 different actuator values.
+	constexpr float HIGH_THRUST_Z_PER_MOTOR{-THRUST_Z_TOTAL / 4 + PITCH_DIFF_PER_MOTOR};
+	constexpr float LOW_THRUST_Z_PER_MOTOR{-THRUST_Z_TOTAL / 4 - PITCH_DIFF_PER_MOTOR};
+	EXPECT_TRUE(is_similar(actuator_sp(0), HIGH_THRUST_Z_PER_MOTOR));
+	EXPECT_TRUE(is_similar(actuator_sp(1), LOW_THRUST_Z_PER_MOTOR));
+	EXPECT_TRUE(is_similar(actuator_sp(2), HIGH_THRUST_Z_PER_MOTOR));
+	EXPECT_TRUE(is_similar(actuator_sp(3), LOW_THRUST_Z_PER_MOTOR));
+
+	for (size_t i{4}; i < ActuatorEffectiveness::NUM_ACTUATORS; ++i) {
+		EXPECT_TRUE(is_similar(actuator_sp(i), 0));
+	}
+}
+
+// This tests that a control setpoint for z-thrust + yaw returns the desired actuator setpoint.
+// This test saturates yaw and demonstrates reduction of thrust for yaw.
+TEST(ControlAllocationSequentialDesaturationTest, AirmodeDisabledReducedThrustAndYaw)
+{
+	ControlAllocationSequentialDesaturation allocator;
+	setup_quad_allocator(allocator);
+	matrix::Vector<float, ActuatorEffectiveness::NUM_AXES> control_sp;
+	// Negative, because +z is "downward".
+	constexpr float DESIRED_THRUST_Z_PER_MOTOR{0.8};
+	constexpr float THRUST_Z_TOTAL{-DESIRED_THRUST_Z_PER_MOTOR * 4};
+	// This is arbitrarily high to trigger strongest possible (saturated) yaw response.
+	constexpr float YAW_CONTROL_SP{1};
+	control_sp(ControlAllocation::ControlAxis::ROLL) = 0;
+	control_sp(ControlAllocation::ControlAxis::PITCH) = 0;
+	control_sp(ControlAllocation::ControlAxis::YAW) = YAW_CONTROL_SP;
+	control_sp(ControlAllocation::ControlAxis::THRUST_X) = 0;
+	control_sp(ControlAllocation::ControlAxis::THRUST_Y) = 0;
+	control_sp(ControlAllocation::ControlAxis::THRUST_Z) = THRUST_Z_TOTAL;
+	allocator.setControlSetpoint(control_sp);
+
+	// Since MC_AIRMODE was not set explicitly, assume airmode is disabled.
+	allocator.allocate();
+
+	const auto &actuator_sp = allocator.getActuatorSetpoint();
+	// In the case of yaw saturation, thrust per motor will be reduced by the hard-coded
+	// magic-number yaw margin of 0.15f.
```

Comment:
@MaEtUgR maybe it's a good time to document this magic number in the algorithm as well!

Reply:
I can only imagine why this is useful. I think @bresch had the full context here.